lib/sanbase/clickhouse/miner_balances_over_time.ex

```diff
@@ -0,0 +1,66 @@
+defmodule Sanbase.Clickhouse.MinerBalancesOverTime do
+  @moduledoc ~s"""
+  Uses ClickHouse to calculate miner balances over time.
+  """
+
+  alias Sanbase.DateTimeUtils
+  require Sanbase.ClickhouseRepo, as: ClickhouseRepo
+
+  @type miner_balances :: %{
+          datetime: DateTime.t(),
+          balance: float
+        }
+
+  @spec miner_balances(
+          DateTime.t(),
+          DateTime.t(),
+          String.t()
+        ) :: {:ok, list(miner_balances)} | {:error, String.t()}
```

Comment:
```suggestion
        ) :: {:ok, list(miner_balance)} | {:error, String.t()}
```

plural is a list(singular) fits better